src/main/resources/controllers-context.xml

```diff
@@ -6,8 +6,24 @@
        http://www.springframework.org/schema/mvc
        https://www.springframework.org/schema/mvc/spring-mvc.xsd">
 
-    <bean class="com.example.dog.DogController">
+    <bean class="com.example.dog.controller.DogController">
+        <constructor-arg ref="jdbcDogDao"/>
     </bean>
+
+    <bean id="inMemoryDogDao" class="com.example.dog.dao.InMemoryDogDAO">
+    </bean>
+
+    <bean id="jdbcDogDao" class="com.example.dog.dao.JdbcDogDAO" primary="true">
+        <constructor-arg ref="dataSource"/>
+    </bean>
+
+    <bean id = "dataSource" class = "org.springframework.jdbc.datasource.DriverManagerDataSource">
+        <property name = "driverClassName" value = "org.h2.Driver"/>
```

Comment:
After you push to master, let's make these configurable properties. Move the values into some kind of `application.properties` and use placeholders here.